koku/masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -125,8 +129,17 @@ def _check_size(self, s3key, check_inflate=False):
         """
         size_ok = False
 
-        s3fileobj = self.s3_client.get_object(Bucket=self.report.get("S3Bucket"), Key=s3key)
-        size = int(s3fileobj.get("ContentLength", -1))
+        try:
+            s3fileobj = self.s3_client.get_object(Bucket=self.report.get("S3Bucket"), Key=s3key)
+            size = int(s3fileobj.get("ContentLength", -1))
+        except ClientError as ex:
+            if ex.response["Error"]["Code"] == "AccessDenied":
+                msg = "Unable to access S3 Bucket {}: (AccessDenied)".format(self.report.get("S3Bucket"))
+                LOG.info(log_json(self.tracing_id, msg, self.context))
+                raise AWSReportDownloaderAccessDenied(msg)
+            msg = f"Error downloading file: Error: {str(ex)}"
+            LOG.error(log_json(self.tracing_id, msg, self.context))
+            raise AWSReportDownloaderError(str(ex))
```

Comment:
Question: Could you explain why this error handling was added? The only time we use `_check_size` is _after_ we have the object. I cannot imagine us downloading the report first and then suddenly we can't download it again...but also, side tangent, and unrelated to this PR, I don't even know why we download it again. We should get the size when we initially download the report and pass it into here.

Reply:
So, I interpreted your original comment to mean: "Is there anywhere else in this file where we could potentially run into an access denied error that is not being covered." 

With that context in mind, my changes here probably make more sense now. 😂 

The changes of someone loosing their permissions between calls is low, but there was a possibility so I wrapped it in a try catch. 

However, after seeing what you linked in your other comment I am now picking up what you mean. You bring up a great point about us only catching the `AWSReportDownloaderNoFileError` in those two places you mentioned, and in hindsight, I think I should just call `AWSReportDownloaderNoFileError` instead of creating a new exception. I personally think access denied would be a perfectly good reason for us to have no file.